src/testdir/test_quickfix.vim

```diff
@@ -6718,4 +6718,24 @@ func Test_hardlink_fname()
   call Xtest_hardlink_fname('l')
 endfunc
 
+func Test_event_QufickFixCmdChanged()
+  let g:count = 0
+  autocmd QuickFixCmdChanged * let g:count = g:count + 1
+
+  let g:l = [{'text': 'foo'}] + repeat([{'text': join(range(30))}], 2)
+  copen
+  call feedkeys(":call setqflist(g:l, 'r')\<CR>", "tx")
+  call assert_equal(1, g:count)
+
+  call feedkeys(":call setqflist(g:l, 'r')\<CR>", "tx")
+  call assert_equal(2, g:count)
+
+  call feedkeys(":call setloclist(0,g:l, 'r')\<CR>", "tx")
+  call assert_equal(3, g:count)
+
+  ccl
+  autocmd! QuickFixCmdChanged
+  unlet g:count
+endfunc
```

Comment:
Can you add a test case for modifying the quickfix list from this autocmd? Also, add a test case for deleting the quickfix or location list from this autocmd.  We have seen issues like this before.

Reply:
> Can you add a test case for modifying the quickfix list from this autocmd?

this will cause autocmd recursive error..